### PR TITLE
{AKS} Support reset default value for loadbalancer profile and natgateway profile

### PR DESCRIPTION
**Pull request by MartinForReal**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
`az aks update --load-balancer-outbound-ports 0`

### General Guidelines
related: #27071

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/aks-preview/azext_aks_preview/tests/latest/test_natgateway.py [hunk 1]
```diff
@@ -58,6 +58,19 @@ def test_empty_arguments(self):
...
 
+    def test_reset_empty_arguments(self):
+        origin_profile = self.nat_gateway_models.ManagedClusterNATGatewayProfile(
+            managed_outbound_ip_profile=self.nat_gateway_models.ManagedClusterManagedOutboundIPProfile(
+                count=1
+            ),
+            idle_timeout_in_minutes=4
+        )
+
+        profile = natgateway.update_nat_gateway_profile(0, None, origin_profile, models=self.nat_gateway_models)
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/tests/latest/test_natgateway.py:69]: `update_nat_gateway_profile` is imported from azure-cli and the code there doesn't yet contain the fix. In addition, even if the fix there is merged into the dev branch, aks-preview's dependency on azure-cli is still at 2.49.0, which does not include fixes.

I think you need to override the implementation here in aks-preview.

File: src/aks-preview/azext_aks_preview/_loadbalancer.py [hunk 2]
```diff
@@ -85,11 +85,11 @@ def create_load_balancer_profile(
...
     profile = ManagedClusterLoadBalancerProfile()
     return configure_load_balancer_profile(
-        managed_outbound_ip_count,
-        managed_outbound_ipv6_count,
+        managed_outbound_ip_count is not None,
+        managed_outbound_ipv6_count is not None,
         outbound_ips,
         outbound_ip_prefixes,
-        outbound_ports,
+        outbound_ports is not None,
```
[the remark this answers is not in the document]
Comment (MartinForReal) [reply src/aks-preview/azext_aks_preview/_loadbalancer.py]: fixed. Thanks!

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 3]
```diff
@@ -659,6 +659,18 @@ def get_load_balancer_managed_outbound_ipv6_count(self) -> Union[int, None]:
         """
         return self.raw_param.get('load_balancer_managed_outbound_ipv6_count')
 
+    def get_load_balancer_outbound_ports(self) -> Union[int, None]:
```
[the remark this answers is not in the document]
Comment (MartinForReal) [reply src/aks-preview/azext_aks_preview/managed_cluster_decorator.py]: fixed. Thanks!

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 4]
```diff
@@ -691,6 +703,32 @@ def get_load_balancer_backend_pool_type(self) -> str:
         # this parameter does not need validation
         return load_balancer_backend_pool_type
 
+    def get_nat_gateway_managed_outbound_ip_count(self) -> Union[int, None]:
```
[the remark this answers is not in the document]
Comment (MartinForReal) [reply src/aks-preview/azext_aks_preview/managed_cluster_decorator.py]: fixed. Thanks!

File: src/aks-preview/azext_aks_preview/tests/latest/test_natgateway.py (hunk 1 above)
Comment (MartinForReal) [reply src/aks-preview/azext_aks_preview/tests/latest/test_natgateway.py:69]: fixed. Thanks!

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 5]
```diff
@@ -3486,7 +3524,13 @@ def check_raw_parameters(self):
...
                 is_different_from_special_default = True
                 break
-
+        if any([
+            self.context.get_load_balancer_managed_outbound_ip_count() is not None,
+            self.context.get_load_balancer_managed_outbound_ipv6_count() is not None,
+            self.context.get_load_balancer_outbound_ports() is not
+            self.context.get_nat_gateway_managed_outbound_ip_count() is None,
+        ]):
+            is_different_from_special_default = True
```
[the remark this answers is not in the document]
Comment (MartinForReal) [reply src/aks-preview/azext_aks_preview/managed_cluster_decorator.py]: fixed. Thanks!

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 6]
```diff
@@ -2691,6 +2695,13 @@ def set_up_network_profile(self, mc: ManagedCluster) -> ManagedCluster:
                 models=self.models.load_balancer_models,
             )
 
+        if self.context.get_nat_gateway_managed_outbound_ip_count() is not None:
+            network_profile.nat_gateway_profile = create_nat_gateway_profile(
+                self.context.get_nat_gateway_managed_outbound_ip_count(),
+                self.context.get_nat_gateway_idle_timeout(),
+                models=self.models.nat_gateway_models,
+            )
+
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:2704]: Perhaps this part can be omitted as it's part of the create process, even if setting it to 0 is ignored, the default value will be backfilled to 0?

**Review by FumingZhang — COMMENTED** [review]
lgtm

Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=86742645&view=results) to validate the change, test passed!

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py (hunk 6 above)
Comment (MartinForReal) [reply src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:2704]: The default value is 1

File: src/aks-preview/HISTORY.rst [hunk 7]
```diff
@@ -13,6 +13,7 @@ Pending
 +++++++
 * Deprecate the alias "-r" of parameter --source-resource-id in `az aks trustedaccess rolebinding create`
 * Refactor azure service mesh related code to meet cli style requirements.
+* Support reset default value for loadbalancer profile and natgateway profile
```
Comment (FumingZhang) [inline src/aks-preview/HISTORY.rst:16]: I think you'll need to rebase from main branch, and update your history note, 2.x.x has been released

**Comment by FumingZhang** [comment]
Requeued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=89894635&view=results)
- test_aks_create_and_update_outbound_ips

**Comment by MartinForReal** [comment]
> Requeued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=89894635&view=results)
> 
> * test_aks_create_and_update_outbound_ips

The test passed.

**Review by FumingZhang — APPROVED** [review]
Syncing changes in official azure-cli [#28273](https://github.com/Azure/azure-cli/pull/28273)

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**